app/src/main/java/com/metrostateics499/vre_app/view/SetNewPasswordActivity.kt

```diff
@@ -0,0 +1,126 @@
+package com.metrostateics499.vre_app.view
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.metrostateics499.vre_app.R
+import com.metrostateics499.vre_app.model.Passing
+
+class SetNewPasswordActivity : AppCompatActivity() {
+
+    private lateinit var newPassword: EditText
+    private lateinit var confirmNewPassword: EditText
+    private lateinit var textEmailUsernameHint: TextView
+    private lateinit var newPasswordContButton: Button
+    private lateinit var invalidNewPassword: TextView
+    private var adminUsername: String = "username"
+    private var username: String = ""
+
+    /**
+     * This overrides the onCreate of this activity
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // This is to hide the action bar
+        supportActionBar?.hide()
+
+        setContentView(R.layout.activity_set_new_password)
+
+        newPassword = findViewById(R.id.new_password)
+        confirmNewPassword = findViewById(R.id.confirm_new_password)
+        newPasswordContButton = findViewById(R.id.new_password_cont_button)
+        textEmailUsernameHint = findViewById(R.id.set_password_email_username_display)
+        invalidNewPassword = findViewById(R.id.invalid_new_password)
+
+        // Get the Email/Username from ForgotPasswordActivity
+        username = intent.getStringExtra("email/username").toString()
+
+        // Display username on user interface xml
+        textEmailUsernameHint.text = username
+
+        // The continue button on user interface
+        newPasswordContButton.setOnClickListener {
+            verifyReplacePassword()
+        }
+    }
+
+    /**
+     * The function to verify the new password and replaces it
+     */
+    private fun verifyReplacePassword() {
+        val inputNewPassword = newPassword.text.toString()
+        val inputConfirmNewPassword = confirmNewPassword.text.toString()
+        val emptyNewPassword = "Password is needed to continue"
+        val doesNotMatchPassword = "Password does not match. Try Again"
+        val lengthNotCorrectPassword = "Password must be 8 to 36 characters"
```

Comment:
Im not sure we need to do this. Might be redundant. We can just use buildString { append("Textstuff") }
We can fix this in next milestone

Reply:
Okay